mujoco_py/mjpid.pyx

```diff
@@ -24,77 +81,191 @@ cdef enum USER_DEFINED_ACTUATOR_PARAMS:
     IDX_DERIVATIVE_GAIN_SMOOTHING = 4,
     IDX_ERROR_DEADBAND = 5,
 
-
 cdef enum USER_DEFINED_CONTROLLER_DATA:
     IDX_INTEGRAL_ERROR = 0,
     IDX_LAST_ERROR = 1,
     IDX_DERIVATIVE_ERROR_LAST = 2,
-    NUM_USER_DATA_PER_ACT = 3,
+    # Needs to be max() of userdata needed for all control modes. 5 needed for cascaded PI
+    NUM_USER_DATA_PER_ACT = 5,
```

Comment:
better to define num_user_data_per_act as max(pid_user_data_max, cascade_pid_user_data_max), each define its own max_enum, global max is the max of both.

Reply:
It would change how memory is allocated for PID vs cascaded though (i.e., they won't be aligned to the maximum if we use local NUM_USER_DATA in each implementation) and also then we cannot support a mix and match of actuator types due to this alignment. what do you think?

Reply:
what I mean is that we still use the same NUM_USER_DATA_PER_ACT to control the memory, but just set it in the code as max of two separate value, so it is more explicit and easier to extend.

Reply:
gotcha. will do that and use `NUM_USER_DATA_PER_ACT` uniformly in the implementation